app/controllers/application_controller.rb

```diff
@@ -1,10 +1,153 @@
 require './config/environment'
-
+require "pry"
 class ApplicationController < Sinatra::Base
 
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "time again"
+  end
+
+  get "/" do
+    erb :index
+  end
+
+  get "/signup" do
+    if logged_in?
+      redirect "/tweets"
+    else
+      erb :"users/create_user"
+    end
+  end
+
+  post "/signup" do
+    if !params[:username].empty? && !params[:email].empty? && !params[:password].empty?
+      @user = User.create(params)
+      session[:id] = @user.id
+
+      redirect "/tweets"
+    else
+      redirect "/signup"
+    end
+  end
+
+  get "/login" do
+    if logged_in?
+      redirect "/tweets"
+    else
+      erb :"users/login"
+    end
+  end
+
+  post "/login" do
+    @user = User.find_by(username: params[:username])
+    if @user && @user.authenticate(params[:password])
+      session[:id] = @user.id
+      redirect "/tweets"
+    else
+      redirect "/login"
+    end
+  end
+
+  get "/tweets/logout" do
+    session.clear
+    redirect "/"
+  end
+
+  get "/tweets/new" do
+    if logged_in?
+      erb :"tweets/create_tweet"
+    else
+      redirect "/login"
+    end
+  end
+
+  post "/tweets" do
+    if !params[:content].empty?
+      @tweet = Tweet.create(content: params[:content], user_id: current_user.id)
+      redirect "/tweets"
+    else
+      redirect "/tweets/new"
+    end
+  end
+
+  get "/tweets" do
+    @tweets = Tweet.all
+    if logged_in?
+      erb :"tweets/tweets"
+    else
+      redirect "/login"
+    end
+  end
+
+  get "/logout" do
+    session.clear
+    redirect "/login"
+  end
+
+  get "/users/:slug" do
+    @user = User.find_by_slug(params[:slug])
+    @tweets = @user.tweets
+    erb :"users/tweets"
+  end
+
+  get "/tweets/:id" do
+    @tweet = Tweet.find_by(id: params[:id])
+
+    if logged_in? && @tweet
+      erb :"tweets/show_tweet"
+    else
+      redirect "/login"
+    end
+  end
+
+  get "/tweets/:id/edit" do
+    @tweet = Tweet.find_by(id: params[:id])
+#    binding.pry
+    if logged_in? && @tweet && current_user.id == session[:id]
+      erb :"tweets/edit_tweet"
+    else
+      redirect "/login"
+    end
+  end
+
+  post "/tweets/:id/edit" do
+    @tweet = Tweet.find(params[:id])
+    if !params[:content].empty?
+      @tweet.update(content: params[:content] )
+      redirect "/tweets"
+    else
+      redirect "/tweets/#{@tweet.id}/edit"
+    end
+  end
+
+  delete "/tweets/:id/delete" do
+    @tweet = Tweet.find(params[:id])
```

Comment:
I suppose you meant `  tweet = current_user.tweets.find_by(id: params[:id])
    if tweet && tweet.destroy
      redirect "/tweets"
    else
      tweet = Tweet.find_by(id: params[:id])
      redirect "/tweets/#{tweet.id}"
    end`
The reason for this is because if tweet has a value of nil, it will not redirect to "tweets/tweet.id" because there is no tweet.id attribute.